Add api doc

YujiOshima opened the pull request:
Fix #150 
We can gen doc by `pkg/api/build.sh` script from api.proto.
I didn't use Kubernetes doc generater https://kubernetes.io/docs/contribute/generate-ref-docs/kubernetes-api/
It may be better to use it in the future.

Signed-off-by: YujiOshima <yuji.oshima0x3fd@gmail.com>

YujiOshima commented:
/assign @jlewi @richardsliu

johnugeorge commented on pkg/api/api.swagger.json:
```diff
@@ -347,6 +356,7 @@
     },
     "/api/Manager/GetWorkerFullInfo": {
       "get": {
+        "summary": "* \nGet a full informations related to specified Workers.\nIt includes Worker Config, HyperParameters and Metrics Logs.",
```
a full informations -> full information

johnugeorge commented on pkg/api/api.swagger.json:
```diff
@@ -390,6 +400,7 @@
     },
     "/api/Manager/GetWorkers": {
       "get": {
+        "summary": "* \nGet a Worker Configs and Statuses from DB by ID of Study, Trial or Worker.",
```
configs and status of a worker

jlewi commented:
Thanks @YujiOshima are instructions for generating the API docs somewhere? If not could we add them?

YujiOshima commented:
@richardsliu @hougangliu Thank you! I updated!
@jlewi I added README https://github.com/kubeflow/katib/blob/c6525022615c7275c6fdac30fda1fc2a87be7944/pkg/api/README.md for API files and docs.

jlewi commented:
/assign @hougangliu 
/unassign @richardsliu  (OOO)

LGTM

@hougangliu Could you take another look please and see if you are satisfied? thanks!

hougangliu commented:
Thanks @YujiOshima . Lgtm except these typo error.

YujiOshima commented:
@hougangliu Thanks! fixed.
